packages/wrangler/src/publish.ts

```diff
@@ -242,6 +242,14 @@ export default async function publish(props: Props): Promise<void> {
       ? `/accounts/${accountId}/workers/services/${scriptName}/environments/${envName}`
       : `/accounts/${accountId}/workers/scripts/${scriptName}`;
 
+    // it can be confusing to see "uploaded <worker>" but fail on the deployment step
+    // when the user is uploading to workers.dev but hasn't registered a subdomain,
+    // so we head off a "subdomain doesn't exist" error early
```

Comment:
This would make all publishes slower. Why can't we capture the specific error when publishing instead, and show the polite message?

Reply:
We can defer the "uploaded worker" message to later, that's fine

Reply:
done